enterprise/server/workflow/service/service.go

```diff
@@ -610,25 +561,104 @@ func (ws *workflowService) getActions(ctx context.Context, wf *tables.Workflow,
 	return filteredActions, nil
 }
 
-func (ws *workflowService) getRepositoryWorkflow(ctx context.Context, id string) (*repositoryWorkflow, error) {
-	app := ws.env.GetGitHubApp()
-	if app == nil {
-		return nil, status.UnimplementedError("GitHub App is not configured")
+func (ws *workflowService) getWorkflowByID(ctx context.Context, workflowID string) (*tables.Workflow, error) {
+	isLegacyWorkflow := !isRepositoryWorkflowID(workflowID)
+	if isLegacyWorkflow {
+		wf := &tables.Workflow{}
+		err := ws.env.GetDBHandle().DB(ctx).Raw(
+			`SELECT * FROM Workflows WHERE workflow_id = ?`,
+			workflowID,
+		).Take(wf).Error
+		if err != nil {
+			if db.IsRecordNotFound(err) {
+				return nil, status.NotFoundError("Workflow not found")
+			}
+			return nil, status.InternalError(err.Error())
+		}
+		return wf, nil
 	}
-	groupID, repoURL, err := parseRepositoryWorkflowID(id)
+
+	// If the workflow ID identifies a GitRepository, look up the GitRepository and construct a synthetic Workflow
+	// from it.
+	groupID, repoURL, err := parseRepositoryWorkflowID(workflowID)
+	rwf, err := ws.getRepositoryWorkflow(ctx, groupID, repoURL)
 	if err != nil {
 		return nil, err
 	}
+
+	return rwf.Workflow, nil
+}
+
+func (ws *workflowService) GetWorkflowByGroupAndRepo(ctx context.Context, groupID string, repoURL string) (*tables.Workflow, error) {
+	parsedRepoURL, err := gitutil.ParseGitHubRepoURL(repoURL)
+	if err != nil {
+		return nil, err
+	}
+
+	rwf, err := ws.getRepositoryWorkflow(ctx, groupID, parsedRepoURL)
+	if err == nil {
+		return rwf.Workflow, nil
+	}
+
+	// If the workflow isn't found in the GitRepositories table, look in the legacy Workflows table
+	wf := &tables.Workflow{}
+	err = ws.env.GetDBHandle().DB(ctx).Raw(`
+		SELECT *
+		FROM Workflows
+		WHERE group_id = ?
+		AND repo_url = ?
+	`, groupID, repoURL).Take(wf).Error
+	if err != nil {
+		if db.IsRecordNotFound(err) {
+			return nil, status.NotFoundErrorf("repo %q not found", repoURL)
+		}
+		return nil, status.InternalErrorf("failed to look up repo %q: %s", repoURL, err)
+	}
```

Comment:
I think we should remove the legacy workflow lookup, and we can tell people that if they want to use the public API then GitHub App authentication is required.

I eventually want to phase out the Workflows table and this is sort of a small step towards that.